### PR TITLE
Implement Asterisk phone provider

### PR DESCRIPTION
**Pull request by vucong2409**
# What this PR does
This PR adds support for Asterisk call center beside Twilio.
## Checklist
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,7 +54,7 @@ ifeq ($(DB),$(SQLITE_PROFILE))
 endif
 
 define run_docker_compose_command
-	$(DOCKER_COMPOSE_ENV_VARS) docker compose -f $(DOCKER_COMPOSE_FILE) $(1)
+	$(DOCKER_COMPOSE_ENV_VARS) docker-compose -f $(DOCKER_COMPOSE_FILE) $(1)
```
Comment (sparfenov) [inline Makefile:57]: https://docs.docker.com/compose/compose-v2/#differences-between-compose-v1-and-compose-v2

Comment (vucong2409) [reply Makefile]: Thanks for pointing that out! I've reverted that change.

**Comment by dzift** [comment]
Hello! can you show an example of a working configuration for testing? When I send alerts, oncall requires twilio to confirm numbers, but I'm trying to integrate with asterisk.

**Comment by badhop** [comment]
> Hello! can you show an example of a working configuration for testing? When I send alerts, oncall requires twilio to confirm numbers, but I'm trying to integrate with asterisk.

Also have the same problem. Need twilio to confirm telephone number.

**Comment by Git-general-InternalIT** [comment]
> > Hello! can you show an example of a working configuration for testing? When I send alerts, oncall requires twilio to confirm numbers, but I'm trying to integrate with asterisk.
> 
> Also have the same problem. Need twilio to confirm telephone number.

In Env Variables -> PHONE_PROVIDER, need set to Asterisk

**Comment by badhop** [comment]
> > > Hello! can you show an example of a working configuration for testing? When I send alerts, oncall requires twilio to confirm numbers, but I'm trying to integrate with asterisk.
> > 
> > 
> > Also have the same problem. Need twilio to confirm telephone number.
> 
> In Env Variables -> PHONE_PROVIDER, need set to Asterisk

Already set. When you try verify phone number it sends request to asterisk, but without verification code.

**Comment by vucong2409** [comment]
Hi guys,
Sorry for the late reply
Currently, I sent a message containing OTP via `alertMessage`  variable (you guy can see that in engine/apps/twilioapp/asterisk_client.py)  and expect Asterisk to have some sort of TTS set up to read it.

For example, I have a dial plan set up like below. I use https://github.com/zaf/asterisk-googletts for TTS.

```
[oncall]
exten => 12345, 1, Answer()
           =>  12345, n, agi(googletts.agi,${alertMessage},en)
           =>  12345, n, Hangup()
```

**Comment by dzift** [comment]
> > > > Hello! can you show an example of a working configuration for testing? When I send alerts, oncall requires twilio to confirm numbers, but I'm trying to integrate with asterisk.
> > > 
> > > 
> > > Also have the same problem. Need twilio to confirm telephone number.
> > 
> > 
> > In Env Variables -> PHONE_PROVIDER, need set to Asterisk
> 
> Already set. When you try verify phone number it sends request to asterisk, but without verification code.

Hello! Did you manage to solve the phone number verification problem?

**Comment by dzift** [comment]
Missing password field for EXTENSION in Env Variables

**Comment by Konstantinov-Innokentii** [comment]
Hi, @vucong2409. First of all - sorry for the late reply. Second - thanks for the PR.
I want to finish work on that to provide asterisk feature for OSS and make phone providers little more "robust".
I started to dive into the code, could you please say what is the purpose of "asterisk_otp" field in used model?

**Comment by Konstantinov-Innokentii** [comment]
Hi everyone. I merged https://github.com/grafana/oncall/pull/1713 with refactoring of phone provider system, which should make implementing asterisk much simpler. Waiting for contributions :)

**Comment by Konstantinov-Innokentii** [comment]
We are still waiting for volunteers to finish this PR, feel free to contact me to sync up on that!

**Comment by joeyorlando** [comment]
closing this PR as it needs to be refactored to use the new phone provider interface. It's not currently on the core team's roadmap to add this provider, but if anyone would like to open a new PR, implemented using the new interface, feel free and we will definitely review 🙂

**Pull request closed without merging**